ui/src/reusable_ui/components/dropdowns/ActionDropdown.tsx

```diff
@@ -0,0 +1,209 @@
+import classnames from 'classnames'
+
+// Components
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import DropdownDivider from 'src/reusable_ui/components/dropdowns/DropdownDivider'
+import DropdownItem from 'src/reusable_ui/components/dropdowns/DropdownItem'
+import DropdownButton from 'src/reusable_ui/components/dropdowns/DropdownButton'
+import FancyScrollbar from 'src/shared/components/FancyScrollbar'
+
+// Types
+import {
+  DropdownMenuColors,
+  ComponentStatus,
+  ComponentColor,
+  ComponentSize,
+  IconFont,
+} from 'src/reusable_ui/types'
+
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+interface Props {
+  children: JSX.Element[]
+  onChange: (value: any) => void
+  actionText: string
+  buttonColor?: ComponentColor
+  buttonSize?: ComponentSize
+  menuColor?: DropdownMenuColors
+  status?: ComponentStatus
+  widthPixels?: number
+  icon?: IconFont
+  wrapText?: boolean
+  customClass?: string
+  maxMenuHeight?: number
+}
+
+interface State {
+  expanded: boolean
+}
+
+@ErrorHandling
+class ActionDropdown extends Component<Props, State> {
+  public static defaultProps: Partial<Props> = {
+    buttonColor: ComponentColor.Default,
+    buttonSize: ComponentSize.Small,
+    status: ComponentStatus.Default,
+    wrapText: false,
+    maxMenuHeight: 250,
+    menuColor: DropdownMenuColors.Sapphire,
+  }
+
+  public static Button = DropdownButton
+  public static Item = DropdownItem
+  public static Divider = DropdownDivider
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      expanded: false,
+    }
+  }
+
+  public render() {
+    const {widthPixels} = this.props
+    const width = widthPixels ? `${widthPixels}px` : '100%'
+
+    this.validateChildCount()
+
+    return (
+      <ClickOutside onClickOutside={this.collapseMenu}>
+        <div className={this.containerClassName} style={{width}}>
+          {this.button}
+          {this.menuItems}
+        </div>
+      </ClickOutside>
+    )
+  }
+
+  private toggleMenu = (): void => {
+    this.setState({expanded: !this.state.expanded})
+  }
+
+  private collapseMenu = (): void => {
+    this.setState({expanded: false})
+  }
+
+  private get containerClassName(): string {
+    const {buttonColor, buttonSize, status, wrapText, customClass} = this.props
+
+    return classnames(
+      `dropdown dropdown-${buttonSize} dropdown-${buttonColor}`,
+      {
+        disabled: status === ComponentStatus.Disabled,
+        'dropdown-wrap': wrapText,
+        [customClass]: customClass,
+      }
+    )
+  }
+
+  private get button(): JSX.Element {
+    const {actionText, status, buttonColor, buttonSize, icon} = this.props
+    const {expanded} = this.state
+
+    return (
+      <DropdownButton
+        active={expanded}
+        color={buttonColor}
+        size={buttonSize}
+        icon={icon}
+        onClick={this.toggleMenu}
+        status={status}
+      >
+        {actionText}
```

Comment:
It seems like this component is pretty much the same as the reusable ui dropdown component other than this here. Is there a reason for making a new component rather than modifying Dropdown to handle this case as well? Also correct me if im wrong and there are more differences than im seeing.

Reply:
yeah, for sure. i believe @alexpaxton wanted to keep the functionality separate, but there is very little difference as you've said -- thoughts @alexpaxton?